packages/chain-adapters/src/evm/bnbsmartchain/BscChainAdapter.test.ts

```diff
@@ -0,0 +1,633 @@
+    })
+
+    it('should return false for an empty address', async () => {
+      const adapter = new bsc.ChainAdapter(makeChainAdapterArgs())
+      const res = await adapter.validateAddress('')
+
+      expect(res).toMatchObject(invalidAddressTuple)
+    })
+
+    it('should return false for an invalid address', async () => {
+      const adapter = new bsc.ChainAdapter(makeChainAdapterArgs())
+      const res = await adapter.validateAddress('foobar')
+
+      expect(res).toMatchObject(invalidAddressTuple)
+    })
+  })
+
+  describe('signTransaction', () => {
+    it('should sign a properly formatted txToSign object', async () => {
+      const balance = '2500000'
+      const httpProvider = {
+        getAccount: jest
+          .fn<any, any>()
+          .mockResolvedValue(makeGetAccountMockResponse({ balance, tokenBalance: '424242' })),
+      } as unknown as unchained.bnbsmartchain.V1Api
+      const args = makeChainAdapterArgs({ providers: { http: httpProvider } })
+      const adapter = new bsc.ChainAdapter(args)
+
+      const tx = {
+        wallet: await getWallet(),
+        txToSign: {
+          addressNList: toAddressNList(adapter.getBIP44Params({ accountNumber: 0 })),
+          value: '0xf0',
+          to: EOA_ADDRESS,
+          chainId: Number(fromChainId(bscChainId).chainReference),
+          data: '0x',
+          nonce: '0x0',
+          gasPrice: '0x12a05f200',
+          gasLimit: '0x5208',
+        },
+      } as unknown as SignTxInput<ETHSignTx>
+
+      await expect(adapter.signTransaction(tx)).resolves.toEqual(
+        '0xf8668085012a05f20082520894d8da6bf26964af9d7eed9e03e53415d37aa9604581f0808193a024e66cc183b9b3e72267ed5db7072789419300d2cb58ccb5f69fc5209f440feaa079cf9b1f0ffb0320d74f809504ced656b064680976ab332bc7a9fb9e4473c8d4',
+      )
+    })
+
+    it('should throw on txToSign with invalid data', async () => {
+      const balance = '2500000'
+      const httpProvider = {
+        getAccount: jest
+          .fn<any, any>()
+          .mockResolvedValue(makeGetAccountMockResponse({ balance, tokenBalance: '424242' })),
+      } as unknown as unchained.bnbsmartchain.V1Api
+      const args = makeChainAdapterArgs({ providers: { http: httpProvider } })
+      const adapter = new bsc.ChainAdapter(args)
+
+      const tx = {
+        wallet: await getWallet(),
+        txToSign: {
+          addressNList: toAddressNList(adapter.getBIP44Params({ accountNumber: 0 })),
+          value: '0x0',
+          to: EOA_ADDRESS,
+          chainId: Number(fromChainId(bscChainId).chainReference),
+          data: 'notHexString',
+          nonce: '0x0',
+          gasPrice: '0x29d41057e0',
+          gasLimit: '0xc9df',
+        },
+      } as unknown as SignTxInput<ETHSignTx>
+
+      await expect(adapter.signTransaction(tx)).rejects.toThrow(/invalid hexlify value/)
+    })
+  })
+
+  // eslint-disable-next-line jest/no-disabled-tests
+  describe.skip('signAndBroadcastTransaction', () => {
```

Comment:
Skipped because of copypasteroo?

Reply:
https://github.com/shapeshift/lib/pull/1197/commits/0ceebad6d160d5cbb19998d7d191448a7cab3f61